pkg/syncer/spec/spec_process.go

```diff
@@ -257,10 +257,16 @@ func (c *Controller) ensureDownstreamNamespaceExists(ctx context.Context, downst
 		shared.NamespaceLocatorAnnotation: string(b),
 	})
 
+	desiredTenantID, err := shared.GetTenantID(desiredNSLocator)
+	if err != nil {
+		return err
+	}
+
 	if upstreamObj.GetLabels() != nil {
 		newNamespace.SetLabels(map[string]string{
 			// TODO: this should be set once at syncer startup and propagated around everywhere.
 			workloadv1alpha1.InternalDownstreamClusterLabel: c.syncTargetKey,
+			shared.TenantIDLabel:                            desiredTenantID,
```

Comment:
We don't update previous namespaces that didn't have this label. This will probably bring hard-to-find bugs for all already-existing downstream namespaces on already-running synced KCP workloads.

Reply:
done. I'm not sure if checking whether GetLabels is nil is necessary or not, all depends on the answer of the question I asked above.